Stock model

rayrobdod opened the pull request:
Basically, Issue #7.

rayrobdod committed:
Add Stock and StockTransaction

rayrobdod committed:
Convert Stock to dumb object

rayrobdod committed:
Stop using nonextant 'Currency' class

I swear that this IDE told me that 'Currency' was a thing earlier.

mikairi commented:
What you are doing is more like Java. The C# way to do properties, for example with `Stock.CurrentPrice`:

``` csharp
public decimal CurrentPrice { get; private set; }
```

If you have a property with a default value like `Note`:

``` csharp
public string Note { get; set; } = "";
```

You don't need those private fields at all. Your code looks cleaner and more concise this way.

Also it's just my personal preference, but `PriceAtTransactionTime` is a little too wordy, while `Price` would practically mean the same thing in the context of a `StockTransaction`. But other than that :+1:

rayrobdod commented:
[Auto-Implemented Properties](https://msdn.microsoft.com/en-us/library/bb384054.aspx). That wasn't the easiest thing to find. Today I learned.

rayrobdod committed:
Using auto-implemented property syntax

Also, shortening the names of two StockTransaction properties.

I'm trusting the documentation when it says "in VS2015, the constructor
can modify a property without a setter." VS2010 does not like it.

mikairi commented:
Since no one else has any problem, I'm merging this.
